### PR TITLE
Remove bluebird dependency

### LINKED ISSUE #529: Switch to native promises instead bluebird
Native promises have increased performance in Node.js 10+, so [bluebird](https://github.com/petkaantonov/bluebird/) doesn't have a significant advantage anymore. Also, bluebird maintainers advice to switch to native promises, when possible. We should experiment with the switch, do some benchmarks and decide whether we should switch now.

If we do the switch, we don't need #524 at all.

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Closes #529

### CODE REVIEW

File: src/network/ClientConnectionManager.ts [hunk 1]
```diff
@@ -632,39 +639,38 @@ export class ClientConnectionManager extends EventEmitter {
...
+                    case AuthenticationStatus.NOT_ALLOWED_IN_CLUSTER:
+                        error = new ClientNotAllowedInClusterError('Client is not allowed in the cluster');
+                        break;
+                    default:
+                        error = new AuthenticationError('Authentication status code not supported. Status: '
+                            + response.status);
                 }
-            });
+                connection.close('Failed to authenticate connection', error);
+                return Promise.reject(error);
```
Comment (mdumandag) [inline src/network/ClientConnectionManager.ts:671]: Minor: Can be replaced with `throw error`

**Review by mdumandag — APPROVED** [review]
Approving with a minor comment

File: src/network/ClientConnectionManager.ts (hunk 1 above)
Comment (puzpuzpuz) [reply src/network/ClientConnectionManager.ts]: Good point. That return was in the original code and it didn't catch my eye. Addressed in 9994e713ce2c4fae6b1008b218e856f7b428a97e

**Comment by puzpuzpuz** [comment]
verify

**Comment by puzpuzpuz** [comment]
verify

**Pull request merged**